shared/fs/footer/upload-container.js

```diff
@@ -33,16 +34,29 @@ const mapDispatchToProps = (dispatch: Dispatch) => ({
   debugToggleShow: getDebugToggleShow(dispatch),
 })
 
-const mergeProps = ({_uploads}, {debugToggleShow}) =>
-  ({
+const mergeProps = ({_uploads, _pathItems}, {debugToggleShow}) => {
+  const folders = _pathItems.filter(path => path.type === 'folder')
```

Comment:
This would end up going through everything. `_pathItems` is a map and you can call `.get` on it to retrieve any item. Why not go through everything in `_uploads.syncingPaths` and check against `_pathItems` to see if anything is a folder?

Reply:
I don't think using`.get` will retrieve the folders in `pathItems`

Reply:
What I meant was getting rid of this `folders` entirely, but instead have something like this below:

```
const filePaths = _uploads.syncingPaths.filter(path => _pathItems.get(path, Constants.unknownPathItem).type !== 'folder')

Reply:
Let me try that